lib/jit_preloader/active_record/base.rb

```diff
@@ -17,13 +17,16 @@ def has_many_aggregate(assoc, name, aggregate, field, default: 0)
       define_method(method_name) do
         self.jit_preload_aggregates ||= {}
 
-        return jit_preload_aggregates[method_name] if jit_preload_aggregates[method_name]
+        return jit_preload_aggregates[method_name] if jit_preload_aggregates.key?(method_name)
         if jit_preloader
           reflection = association(assoc).reflection
           primary_ids = jit_preloader.records.collect{|r| r[reflection.active_record_primary_key] }
           klass = reflection.klass
 
-          preloaded_data = Hash[klass
+          association_scope = klass
+          association_scope = association_scope.instance_exec(&reflection.scope).reorder(nil) if reflection.scope
```

Comment:
I believe `reflection.scope` is present all of the time, but I may be wrong, so I've applied it conditionally. 
Further, if the scope applies an order, that order does not need to be respected for this since we will be doing a group and aggregate, thus we will just remove it.